in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,270 @@
+
+That is,
+
+```math
+\text{new fee asset price modifier} := \text{old fee asset price modifier} + \text{fee asset price modifier delta}
+```
+
+The new `fee_asset_price_modifier_delta` is capped at (+/-) `MAXIMUM_FEE_ASSET_PRICE_PERCENT_CHANGE_PER_BLOCK` * `FEE_ASSET_PRICE_UPDATE_FRACTION` / 100.
+
+### `base_fee_wei_per_mana_congestion_multiplier`
+
+First we compute the excess mana in the current block by considering the parent mana spent and excess mana.
+
+```math
+\text{excess mana} = \begin{cases} 
+0 & \text{if } \text{parent.excess} + \text{parent.spent} < \text{TARGET MANA PER BLOCK} \\
+\text{parent.excess} + \text{parent.spent} - \text{TARGET MANA PER BLOCK} & \text{otherwise}
+\end{cases}
+```
+
+```math
+\begin{aligned}
+\text{base fee wei per mana congestion multiplier} &= \text{MINIMUM CONGESTION MULTIPLIER} * \text{exp}\left(\frac{\text{excess mana}}{\text{CONGESTION MULTIPLIER UPDATE FRACTION}}\right)
+\end{aligned}
+```
+
+## Deriving the base fee
+
+When a proposer is building a block, it calculates:
+```math
+\begin{aligned}
+\text{block cost in wei} &= \text{execution cost in wei} + \text{data cost in wei} \\
+\text{block cost in wei/mana} &= \frac{\text{block cost in wei}}{\text{TARGET MANA PER BLOCK}} + \text{proving cost wei per mana} \\
+\text{base fee in wei/mana} &= \text{block cost in wei/mana} * \text{base fee wei per mana congestion multiplier} \\
+\text{base fee in fee asset/mana} &= \text{base fee in wei/mana} * \text{fee asset per wei}
+\end{aligned}
+```
+
+This final value is the `base_fee_asset_per_mana` field in the L2 block header.
+
+## The cost of a transaction in the fee asset
+
+The amount of mana a transaction consumes is:
+
+```math
+\text{mana}_{tx} = \text{OVERHEAD MANA PER TX} + \sum_{op \in tx} \text{mana per operation}(op)
+```
+
+Therefore, the cost of a transaction's proposal, data publication, and verification at time $i$ is, denominated in the fee asset:
+
+```math
+\text{fee asset}_{tx} = \text{mana}_{tx} * \text{base fee asset per mana}
+```
+
+## Distribution of the fees
+
+When a transaction is included in a block in the pending chain, the L2 balance of the fee asset for the `fee_payer` of the transaction is reduced by the transactions's fee asset cost.
+
+When an epoch is proven, the fees for the epoch are paid out to each proposer for the block(s) they proposed.
+
+A percentage of each block's fee asset fees are paid to the prover of the epoch. The percentage is determined based on the quote submitted by the prover for the epoch.
+
+This quote is submitted by the prover in epoch `i+1` to prove epoch `i`, and is thus claimed/submitted to L1 by a proposer in epoch `i+1`.
+
+In order to prevent a prover bribing a proposer into accepting a quote with an extremely high fee, the maximum fee a prover can submit for an epoch proof quote is `MAXIMUM_EPOCH_PROOF_QUOTE_FEE`.
```

Comment:
We're thinking about this and the best current solution is for `MAXIMUM_EPOCH_PROOF_QUOTE_FEE` to track the fair price of proving, such that sequencers should not "care" which quote they pick. They should just care that they picked a quote. 

Sequencers don't get rewarded for picking a quote.

Reply:
If we did that, maybe we can change the semantics of the quote that provers send, so it's a `minimumBasisPointsFee` rather than a `basisPointsFee`, and the protocol dictates what they are actually paid based on this fair price?

Reply:
My thought here was that proposers in epoch `N` could signal the max fee they're willing to accept, then we take the average in the rollup contract, and the claimed bid must be less than it.

Reply:
This sounds like a fundamental problem with our incentive mechanisms? 
Or, at least, shouldn't this doc have more revisions before it can be merged?

Reply:
I missed this thread. I do think sequencers need to set in their proposal the minimum % fees they will accept for their block in epoch N. The proof quote in epoch N+1 must meet this constraint.

Reply:
I think its fine in general for the sequencer in epoch N +1 to not be incentivised with a fee. Their incentive is to pick a proof quote that is most likely to succeed so their block in epoch N + 1 does not get re-orged.

Reply:
How about then we axe this sentence altogether and add this "average" functionality in a followup?